### PR TITLE
arch: added requested parentheses verifying the lack of ambiguities 

### PR DESCRIPTION
**Pull request by DeHess**
Fix coding guideline MISRA C:2012 Rule 12.1 in arch:

> The precedence of operators within expressions should be made explicit.

This PR is part of the enhancement issue https://github.com/zephyrproject-rtos/zephyr/issues/48002 which port the coding guideline fixes done by BUGSENG on the https://github.com/zephyrproject-rtos/zephyr/tree/v2.7-auditable-branch back to main

The commit in this PR is a subset of the original auditable-branch commit:
https://github.com/zephyrproject-rtos/zephyr/commit/29155bdd6c8f8c2c0bdf0562191b4b968b3a818a

### CODE REVIEW

**Review by jhedberg — DISMISSED** [review]

**Comment by DeHess** [comment]
I'm getting a warning due to the fact that sizeof is being called on an array but I'm not sure whether I should change it to ARRAY_SIZE, that should probably be a different commit.

**Comment by jhedberg** [comment]
> I'm getting a warning due to the fact that sizeof is being called on an array but I'm not sure whether I should change it to ARRAY_SIZE, that should probably be a different commit / PR.

Yes, looks like `ARRAY_SIZE()` should be used. Split that into a separate commit (can still be the same PR). Thanks.

**Comment by DeHess** [comment]
> > I'm getting a warning due to the fact that sizeof is being called on an array but I'm not sure whether I should change it to ARRAY_SIZE, that should probably be a different commit / PR.
> 
> Yes, looks like `ARRAY_SIZE()` should be used. Split that into a separate commit (can still be the same PR). Thanks.

It is done

File: arch/x86/zefi/zefi.c [hunk 1]
```diff
@@ -140,7 +140,7 @@ uintptr_t __abi efi_entry(void *img_handle, struct efi_system_table *sys_tab)
 
 	efi_prepare_boot_arg();
 
-	for (int i = 0; i < sizeof(zefi_zsegs)/sizeof(zefi_zsegs[0]); i++) {
+	for (int i = 0; i < (ARRAY_SIZE(zefi_zsegs)/sizeof(zefi_zsegs[0])); i++) {
```
Comment (jhedberg) [inline arch/x86/zefi/zefi.c:143]: This is not correct. The `ARRAY_SIZE()` macro itself does the division by `array[0]`, so you're now doing it twice here, which in turn will give the wrong result.

File: arch/x86/zefi/zefi.c [hunk 2]
```diff
@@ -150,7 +150,7 @@ uintptr_t __abi efi_entry(void *img_handle, struct efi_system_table *sys_tab)
 		}
 	}
 
-	for (int i = 0; i < sizeof(zefi_dsegs)/sizeof(zefi_dsegs[0]); i++) {
+	for (int i = 0; i < (ARRAY_SIZE(zefi_dsegs)/sizeof(zefi_dsegs[0])); i++) {
```
Comment (jhedberg) [inline arch/x86/zefi/zefi.c:153]: Same here.

**Review by jhedberg — CHANGES_REQUESTED** [review]

File: arch/x86/zefi/zefi.c (hunk 1 above)
Comment (andyross) [reply arch/x86/zefi/zefi.c:143]: Does it even work?  This isn't Zephyr code, it's an EFI bootloader for Zephyr and last I checked (I wrote it originally, but it's been a **long** time) it isn't building against the Zephyr include directories.

**Pull request closed without merging**